### ISSUE TITLE
Question about pointcloud sampler

### ISSUE THREAD

**Issue by DCNSW** [Na-Z/attMPTI#19]
The code [line](https://github.com/Na-Z/attMPTI/blob/main/dataloaders/loader.py#L58) shows that the point clouds sampled for the query set are decided by the groundtruth labels when testing. I don‘t think this is a reasonable experimental design.

Furthermore, why the point cloud in the query set must contain the target class when testing rather than evaluation?
Thanks, @Na-Z

**Comment by Chengyuelan** [comment]
> Why the point cloud in query set must contain the target class? The [code](https://github.com/Na-Z/attMPTI/blob/main/dataloaders/loader.py#L58) also shows that the point clouds sampled for query set are decided by the groundtruth labels. Why is that correct? @Na-Z

@DCNSW I would like to ask, have you implemented part of the code for visualizing the output result?

**Comment by DCNSW** [comment]
I'm consulting the author for the detail of experimental setting in the paper.

**Comment by cemunds** [comment]
Hi, I came across this issue as well. The sampler is using ground truth information to sample a point cloud for the query set that is guaranteed to contain one of the target classes in the support set. In a real-world scenario, it could be possible that the query point cloud does not contain any of the classes in the support set. In theory, one would need to test against multiple query point clouds per training step, where some queries may contain the target classes and some don't. However, testing against a large number of query point clouds on each single step doesn't seem feasible. Maybe one could sample one positive query, that contains the target class and one negative query that doesn't, similar to triplet mining when training siamese neural networks.

As this is basically the first and currently only project that deals with few-shot segmentation for point clouds, I couldn't find another reference implementation to check how they are dealing with this issue. Does anyone know how few-shot segmentation methods for images are approaching this?
